exonum-java-binding/core/rust/exonum-java/src/node.rs

```diff
@@ -70,7 +70,13 @@ fn create_blockchain(
 
     let blockchain = Blockchain::new(database, keypair, api_sender);
 
-    let supervisor_service = supervisor_service();
+    let supervisor_mode = &config
+        .run_config
+        .node_config
```

Comment:
Why don't you take node_config from above? And, possibly, move the remaining extraction code inside supervisor_service?

Reply:
Because `node_config` from above isn't `config.run_config.node_config`. It's exonum core's internal node configuration which does not have supervisor mode option. ECR-4116 created to address API inconvenience that forced this.